WebDriverAgentLib/Commands/FBElementCommands.m

```diff
@@ -608,38 +608,57 @@ + (NSArray *)routes
 #if !TARGET_OS_TV
 static const CGFloat DEFAULT_OFFSET = (CGFloat)0.2;
 
-+ (id<FBResponsePayload>)handleWheelSelect:(FBRouteRequest *)request
-{
-  FBElementCache *elementCache = request.session.elementCache;
-  XCUIElement *element = [elementCache elementForUUID:(NSString *)request.parameters[@"uuid"]];
-  if ([element.lastSnapshot elementType] != XCUIElementTypePickerWheel) {
-    return FBResponseWithStatus([FBCommandStatus invalidArgumentErrorWithMessage:[NSString stringWithFormat:@"The element is expected to be a valid Picker Wheel control. '%@' was given instead", element.wdType] traceback:[NSString stringWithFormat:@"%@", NSThread.callStackSymbols]]);
-  }
-  NSString* order = [request.arguments[@"order"] lowercaseString];
-  CGFloat offset = DEFAULT_OFFSET;
-  if (request.arguments[@"offset"]) {
-    offset = (CGFloat)[request.arguments[@"offset"] doubleValue];
-    if (offset <= 0.0 || offset > 0.5) {
-      return FBResponseWithStatus([FBCommandStatus invalidArgumentErrorWithMessage:[NSString stringWithFormat:@"'offset' value is expected to be in range (0.0, 0.5]. '%@' was given instead", request.arguments[@"offset"]] traceback:[NSString stringWithFormat:@"%@", NSThread.callStackSymbols]]);
+  + (id<FBResponsePayload>)handleWheelSelect:(FBRouteRequest *)request
+  {
+    NSNumber *counter = 0;
+    FBElementCache *elementCache = request.session.elementCache;
+    XCUIElement *element = [elementCache elementForUUID:(NSString *)request.parameters[@"uuid"]];
+    if ([element.lastSnapshot elementType] != XCUIElementTypePickerWheel) {
+      return FBResponseWithStatus([FBCommandStatus invalidArgumentErrorWithMessage:[NSString stringWithFormat:@"The element is expected to be a valid Picker Wheel control. '%@' was given instead", element.wdType] traceback:[NSString stringWithFormat:@"%@", NSThread.callStackSymbols]]);
     }
+    NSString* order = [request.arguments[@"order"] lowercaseString];
+    CGFloat offset = DEFAULT_OFFSET;
+    if (request.arguments[@"offset"]) {
+      offset = (CGFloat)[request.arguments[@"offset"] doubleValue];
+      if (offset <= 0.0 || offset > 0.5) {
+        return FBResponseWithStatus([FBCommandStatus invalidArgumentErrorWithMessage:[NSString stringWithFormat:@"'offset' value is expected to be in range (0.0, 0.5]. '%@' was given instead", request.arguments[@"offset"]] traceback:[NSString stringWithFormat:@"%@", NSThread.callStackSymbols]]);
+      }
+    }
+    BOOL isSuccessful = false;
+    NSError *error;
+    if (request.arguments[@"value"]) {
```

Comment:
Consider flipping this condition and turning it into a guard. long blocks are hard to read